src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNet/AspNetCoreMvc2Integration.cs

```diff
@@ -283,91 +177,121 @@ public AspNetCoreMvc2Integration(object actionDescriptor, object httpContext)
             TargetMaximumVersion = Major2)]
         public static void Rethrow(object context, int opCode, int mdToken)
         {
-            AspNetCoreMvc2Integration integration = null;
-            const string methodName = nameof(Rethrow);
-
-            if (context == null)
-            {
-                // Every rethrow method in every v2.x returns when the context is null
-                // We need the type of context to call the correct method as there are 3
-                // Remove this when we introduce the type arrays within the profiler
-                return;
-            }
+            string methodDef = $"{ResourceInvoker}.{nameof(Rethrow)}({context?.GetType().FullName} context)";
+            var shouldTrace = Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationName);
+            MethodBase instrumentedMethod;
 
             try
             {
-                if (context.TryGetPropertyValue("HttpContext", out object httpContext))
-                {
-                    if (httpContext.TryGetPropertyValue("Items", out IDictionary<object, object> contextItems))
-                    {
-                        integration = contextItems?[HttpContextKey] as AspNetCoreMvc2Integration;
-                    }
-                }
+                instrumentedMethod = Assembly.GetCallingAssembly().ManifestModule.ResolveMethod(mdToken);
             }
             catch (Exception ex)
             {
-                Log.ErrorExceptionForFilter($"Error accessing {nameof(AspNetCoreMvc2Integration)}.", ex);
+                // profiled app will not continue working as expected without this method
+                Log.ErrorException($"Error retrieving {methodDef}", ex);
+                throw;
             }
 
-            Action<object> rethrow;
+            WebServerIntegrationContext integration = null;
 
-            try
-            {
-                rethrow = RethrowAccess.GetInterceptedMethod(
-                    assembly: Assembly.GetCallingAssembly(),
-                    owningType: ResourceInvoker,
-                    returnType: Interception.VoidType,
-                    methodName: methodName,
-                    generics: Interception.NullTypeArray,
-                    parameters: Interception.ParamsToTypes(context));
-            }
-            catch (Exception ex)
+            if (shouldTrace)
             {
-                // profiled app will not continue working as expected without this rethrow method
-                Log.ErrorException($"Error calling {ResourceInvoker}.{methodName}(object context)", ex);
-                throw;
+                var httpContextResult = context.GetProperty("HttpContext");
+
+                if (httpContextResult.HasValue)
+                {
+                    integration = WebServerIntegrationContext.RetrieveFromHttpContext(httpContextResult.Value);
```

Comment:
Does this `integration` variable get used anywhere besides to log an error?

Reply:
No but I do plan to nest a scope in here depending on feedback from Andrew